custom_components/sscpoe/protocol.py

```diff
@@ -440,22 +443,92 @@ def get_host_ip():
+    """Return IPv4 address for interface name (Linux best-effort)."""
+    if platform.system() != "Linux":
+        return None
+    try:
+        import fcntl  # Linux only
+    except Exception:
+        return None
+
+    SIOCGIFADDR = 0x8915
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        ifreq = struct.pack("256s", ifname[:15].encode("utf-8"))
+        res = fcntl.ioctl(s.fileno(), SIOCGIFADDR, ifreq)
+        ip = socket.inet_ntoa(res[20:24])
+        if ip and ip != "0.0.0.0":
+            return ip
+        return None
+    except OSError as e:
+        LOGGER.debug(f"_get_ipv4_for_interface({ifname}) failed: {e}")
+        return None
+    finally:
+        s.close()
+
 
-def SSCPOE_local_send(dt):
+def _try_bindtodevice(sock: socket.socket, ifname: str) -> bool:
+    """Try SO_BINDTODEVICE; may require privileges. Best-effort."""
+    if platform.system() != "Linux":
+        return False
+    try:
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_BINDTODEVICE, ifname.encode("utf-8"))
+        return True
+    except PermissionError:
+        LOGGER.warning(
+            f"SO_BINDTODEVICE({ifname}) not permitted; continue with IP_MULTICAST_IF only"
+        )
+        return False
+    except OSError as e:
+        LOGGER.debug(f"SO_BINDTODEVICE({ifname}) failed: {e}")
+        return False
+
+
+def SSCPOE_local_send(dt, bind_interface: str | None = None, ttl: int | None = None):
+    # Use module-level defaults if caller didn't provide explicit values
+    if bind_interface is None:
+        bind_interface = bind_interface_default
+    if ttl is None:
+        ttl = mcast_ttl_default
     MCAST_GRP = "239.0.0.100"
     MCAST_PORT = 10086
 
-    host = get_host_ip()  # socket.gethostbyname(socket.gethostname())
+    # Select source IPv4:
+    # - bind_interface None/"default": old behaviour (default-route derived host IP)
+    # - bind_interface "<ifname>": use that interface IPv4 to force multicast egress
+    if bind_interface and bind_interface != "default":
+        host = _get_ipv4_for_interface(bind_interface)
+        if not host:
+            raise RuntimeError(
+                f"Interface '{bind_interface}' has no IPv4 address (or not accessible)"
+            )
+    else:
+        host = get_host_ip()  # socket.gethostbyname(socket.gethostname())
 
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
     try:
         sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     except AttributeError:
         pass
-    sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, 2)
+
+    # TTL: keep existing default unless explicitly requested
+    if ttl is not None:
+        if int(ttl) < 1 or int(ttl) > 255:
+            raise ValueError(f"Invalid TTL={ttl}, expected 1..255")
+        sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, int(ttl))
+    else:
+        sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, 2)

```

Comment:
int(ttl) is redundant.
```suggestion
    if ttl is None:
        ttl = 2
    if ttl < 1 or ttl > 255:
        raise ValueError(f"Invalid TTL={ttl}, expected 1..255")
    sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, ttl)
```